src/misc/lv_math.h

```diff
@@ -51,6 +51,23 @@ static inline LV_ATTRIBUTE_FAST_MEM int16_t lv_trigo_cos(int16_t angle)
     return lv_trigo_sin(angle + 90);
 }
 
+/**
+ * Compute the division by 255 without multiplication or divide
+ * @param x The number to divide by 255. Limited to 0-65535, use LV_UDIV255 if the numerator is bigger.
+ * @return x/255 floor rounded.
+ */
+static inline LV_ATTRIBUTE_FAST_MEM uint8_t lv_udiv_255(uint16_t x)
+{
+    /* Mathematical trick here:
+        Since (2^n - 1) = (a - b), and since (a-b)(a+b) = a^2 - b^2, then
+             x / (2^n - 1) = x * (2^n + 1) / [(2^n + 1)*(2^n - 1)] = (x*2^n + x) / (2^(2n) - 1) ~= (x<<n + x) >> (2*n)
+
+        Since 2^(2n) - 1 ~= 2^2n, we commit a smaller error than if we divided by shifting right initially by n.
+        So, for example, for x in [0-65536] range, and dividing by 255, this approximation gives no error */
+    uint32_t num = ((uint32_t)x << 8) + x;
+    return (uint8_t)(num >> 16);
```

Comment:
Is it really faster than `LV_UDIV255`? Here there are 2 shifts and 1 addition.

`LV_UDIV255` is 1 multiplication (1 instruction) and 1 shift.

Reply:
On ESP32, mul is 2 cycles, while add is one cycle, shift is one cycle, so it ends up the same. Div is 4 cycles BTW, so the gain is low anyway.

On ARM cortex M0, multiply are 32 cycles while add & shift are only one cycle.
On ARM cortex M3, multiply takes from 1 to 3 cycles (add & shift = 1 cycle). So either it's faster or slower depends on the chip implementation.
On ARM cortex M7, same as M3/M4 here.

So in the end, if it's faster, it's by a very short margin. I wonder what the compiler is doing for a plain /255 in the code...

Reply:
Oh, I thought mul is 1 i.c. on every modern MCU. Thanks.

> I wonder what the compiler is doing for a plain /255 in the code...

Probably compiler/optimization dependent. 

Anyway, let's leave your option too!

Reply:
The gain does not justify the maintenance burden, so using LV_UDIV255 now.